src/components/Select/Select.component.tsx

```diff
@@ -64,157 +64,169 @@ import {
+            setValue(option.label ?? option.value);
+            setSelectedIndex(active);
+            inputRef.current.blur();
+            onSelectOptionHandler(event, option);
+            setIsOpen(false);
+            return;
+        }
+        if (event.key === 'ArrowUp') {
+            if (active === 0) {
+                return;
+            }
+            setActive((prev) => prev - 1);
+            listRef.current.scrollBy(0, -Constants.ITEM_HEIGHT);
+            return;
+        }
+        if (event.key === 'ArrowDown') {
+            if (active === options.length - 1) {
+                return;
+            }
+            setActive((prev) => prev + 1);
+            listRef.current.scrollBy(0, Constants.ITEM_HEIGHT);
+        }
+    };
 
-	/**
+    /**
 	 * On clicking anywhere other than `input field` or `trailing icon` the dropdown closes
 	 */
-	const onDropDownCloseHandler = (e: MouseEvent) => {
-		if (e.target === inputRef.current || e.target === trailingIconRef.current)
-			return;
-		setIsOpen(false);
-	};
+    const onDropDownCloseHandler = (e: MouseEvent) => {
+        if (e.target === inputRef.current || e.target === trailingIconRef.current) {
+            return;
+        }
+        setIsOpen(false);
+    };
 
-	/**
+    /**
 	 * On the user selecting the option the value/label of the menu item is set to the input field
 	 */
-	const onUserSelectHandler = (
-		e: React.MouseEvent<HTMLLIElement, MouseEvent>,
-		option: ListItemType,
-		index: number
-	) => {
-		setActive(index);
-		setSelectedIndex(index);
-		setValue(option.label ?? option.value);
-		onSelectOptionHandler(e, option);
-	};
+    const onUserSelectHandler = (
+        e: React.MouseEvent<HTMLLIElement, MouseEvent>,
+        option: ListItemType,
+        index: number,
+    ) => {
+        setActive(index);
+        setSelectedIndex(index);
+        setValue(option.label ?? option.value);
+        onSelectOptionHandler(e, option);
+    };
 
-	// On clicking anywhere other than the input field the dropdown closes
-	useEffect(() => {
-		document.body.addEventListener('click', onDropDownCloseHandler);
+    // On clicking anywhere other than the input field the dropdown closes
+    useEffect(() => {
+        document.body.addEventListener('click', onDropDownCloseHandler);
 
-		return () => {
-			document.body.removeEventListener('click', onDropDownCloseHandler);
-		};
-	}, []);
+        return () => {
+            document.body.removeEventListener('click', onDropDownCloseHandler);
+        };
+    }, []);
 
-	/**
+    /**
 	 * If 'isOpen' is true and 'value' is empty, then set the active index to 0 and scroll the list to (0,0)
 	 * else, set the active index to selected item index
 	 */
-	useEffect(() => {
-		if (isOpen) {
-			if (value === '') {
-				setActive(0);
-				listRef.current.scrollTo(0, 0);
-			} else setActive(selectedIndex);
-		}
-	}, [isOpen]);
+    useEffect(() => {
+        if (isOpen) {
+            if (value === '') {
```

Comment:
```suggestion
            if (!value) {
```

Reply:
need value to be empty string. otherwise it will also work for null/undefined